src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -187,6 +208,19 @@ class ObserveFields<VolumeDim, ObservationValueTag, tmpl::list<Tensors...>,
     const std::string element_name =
         MakeString{} << ElementId<VolumeDim>(array_index) << '/';
 
+    // calculate the interpolation matrices if requested. Else, interp_matrices
+    // will not be default constructed and therefore skipped by applyMatrices.
+    std::array<Matrix, VolumeDim> interp_matrices{};
+    if (interpolation_mesh_) {
+      const auto slices = mesh.slices();
+      const auto interpolation_slices = interpolation_mesh_->slices();
+      for (size_t i = 0; i < VolumeDim; i++) {
+        const auto target_points =
+            Spectral::collocation_points(interpolation_slices.at(i));
+        interp_matrices.at(i) =
+            Spectral::interpolation_matrix(slices.at(i), target_points);
+      }
+    }
     // Remove tensor types, only storing individual components.
     std::vector<TensorComponent> components;
```

Comment:
Would you multiply `components.reserve` by `interpolation_mesh->extents() / mesh.extents()` to account for the different size required for the interpolating mesh?

Reply:
The `reserve` only deals with the number of components, not the number of grid points per component

Reply:
ah of course